### ISSUE TITLE
SubjectPresent.deferred=true does not take effect

### ISSUE THREAD

**Issue by jhuang82cn** [schaloner/deadbolt-2-java#25]
Hi, there

I met a problem where `SubjectPresent.deferred=true` does not take effect when running Deadbolt 2.5 with Play 2.5.x

Through debugging I found that the problem is probably caused by AbstractDeadboltAction.java line 112:

```
          if (isDeferred(ctx))
            {
                result = getDeferredAction(ctx).call(ctx);
            }
            else if (!ctx.args.containsKey(IGNORE_DEFERRED_FLAG)
                    && annClass.isAnnotationPresent(Deferrable.class) // <- THIS LINE
                    && (Boolean)annClass.getMethod("deferred").invoke(configuration))
            {
                defer(ctx,
                      this);
                result = delegate.call(ctx);
            }
            else
            {
                result = execute(ctx);
            }
```

In runtime I found the `annClass` is a Java `Proxy` and `annClass.isAnnotationPresent(Deferrable.class)` always returns false.

**Comment by schaloner** [comment]
Thanks, I'll update this in release 2.5.2.

**Comment by mkurz** [comment]
I can confirm this bug, fix is here: #57
However in future versions of deadbolt I would remove the deferred feature and replace with something better. I will open an issue which explains why and how we can achieve that.

**Comment by mkurz** [comment]
This is fixed and will land in 2.5.7 and 2.6.4
